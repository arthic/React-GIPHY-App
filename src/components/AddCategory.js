import React, {useState} from 'react'
import PropTypes from 'prop-types'

export const AddCategory = ({setCategories}) => {
	/* Deja el string vacio para que no te marque error ya que
	si no hay comillas recibe undefined*/
	const [inputValue, setInputValue] = useState('')
	const handleInputChange = (e) => {
		// console.log(e.target.value);
		setInputValue(e.target.value)
		console.log('Cambio el input');
	}
	const handleSubmit = (e) => {
		/*Comportamiento por defecto del formulario es la
		recarga de la pagina*/
		e.preventDefault()
		console.log('handleSubmit', inputValue);

		if (inputValue.trim().length > 2) {
			// Se llama como callback que tiene el estado anterior
			setCategories(cats => [inputValue, ...cats])
			setInputValue('')
		}
	}
	/* const handleFocusChange = (e) => {
		setInputValue(e.target.value)
		// console.log(valorDefault);
		// setInputValue(e.target.value='')
		console.log(e.target.value);
		if (e.target.value === 'Busca un GIF!!!') {
			setInputValue(e.target.value='')
		}
		console.log(e.target.value);
	}*/
	return (
		/* El fragment "<>" agrupa elementos HTML o JSX, al usar
		un Form ya se esta agrupando el elemento, por lo tanto ya
		no es necesario el fragment*/
		<form onSubmit={handleSubmit}>
			<div className="search-contend">
				<i className="fas fa-search"></i>
				<input
					type="text"
					value={inputValue}
					onChange={handleInputChange}
					// onFocus={handleFocusChange}
				/>
			</div>
		</form>
	)
}
AddCategory.propTypes = {
	setCategories: PropTypes.func.isRequired
}
export default AddCategory
