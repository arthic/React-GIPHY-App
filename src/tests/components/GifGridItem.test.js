import React from 'react'
import {shallow} from 'enzyme'
import '@testing-library/jest-dom'
import {GifGridItem} from '../../components/GifGridItem'

describe('Test de GifGridItem', () => {
	const title = 'Un t√≠tulo'
	const url = 'https://localhost/algo.jpg'
	const wrapper = shallow(<GifGridItem title={title} url={url}/>)

	test('Despliege de <GifGridItem /> correcto', () => {
		expect(wrapper).toMatchSnapshot()
	})
	test('El parrafo debe tener el title', () => {
		const p = wrapper.find('p')
		expect( p.text().trim()).toBe(title)
	})
	test('la imagen debe contener las props alt e img correspondientes', () => {
		const img = wrapper.find('img')
		expect( img.prop('src')).toBe(url)
		expect( img.prop('alt')).toBe(title)
	})
	test('Debe contener la clase de css "animate__fadeIn"', () => {
		const div = wrapper.find('div')
		const class_name = div.prop('className')
		expect( class_name.includes('animate__fadeIn')).toBe(true)
	})
});

