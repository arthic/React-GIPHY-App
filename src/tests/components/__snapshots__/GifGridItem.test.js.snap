// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test de GifGridItem Despliege de <GifGridItem /> correcto 1`] = `
<div
  className="card animate__animated animate__fadeIn"
>
  <img
    alt="Un título"
    src="https://localhost/algo.jpg"
  />
  <p>
    Un título
  </p>
</div>
`;
